src/Authenticator/TokenAuthenticator.php

```diff
@@ -118,4 +119,14 @@ public function authenticate(ServerRequestInterface $request, ResponseInterface
 
         return new Result($user, Result::SUCCESS);
     }
+
+    /**
+     * No-op method.
+     *
+     * @param \Psr\Http\Message\ServerRequestInterface $request A request object.
+     * @return void
+     */
+    public function unauthorizedChallenge(ServerRequestInterface $request)
+    {
```

Comment:
We could throw a UnauthorizedException. I added it because the PersistenceInterface requires implementing this method. I added PersistenceInterface to TokenAuthenticator so that identity is not persisted to session when authentication is done use TokenAuthenticator / TokenAuthenticator.